routes/printing.py

```diff
@@ -84,8 +70,23 @@ def print_documents():
                     lectures=lectures)
             sqla.session.add(dep)
             db.accounting.log_deposit(dep, user, data['cash_box'])
+            if documents:
+                name = data['cover_text'].split(' ')[0]
+                for _ in config.print_documents(
+                    doc_paths=[document_path(doc.id) for doc in documents],
+                    cover_text=data['cover_text'],
+                    printer=data['printer'],
+                    user=user.username,
+                    usercode=config.PRINTER_USERCODES[data['cash_box']],
+                    job_title="Odie-Druck für {} [{} Seiten]".format(name, num_pages)):
+                        pass
         sqla.session.commit()
+        yield('accounting succeeded', '')
+    except NonConfidentialException as e:
+        sqla.session.rollback()
+        raise NonConfidentialException(str(e)) from e
     except Exception as e:
```

Comment:
What's up with this Exception flow? Why bother catching `NonConfidentialException` if it is never raised? Why bother catching it specifically at all if you're going to do the same thing for every other exception?